### PR TITLE
Simplify release process and push everyone to 2.0

### PR DESCRIPTION
**Pull request by bmw**
This PR largely undoes these PRs:

* https://github.com/certbot/certbot/pull/9532
* https://github.com/certbot/certbot/pull/9506
* https://github.com/certbot/certbot/pull/9467
* https://github.com/certbot/certbot/pull/9400

I left some of the logic in `tools/finish_release.py` around because I think it may be useful in the future. I tested this script successfully by just printing the `snapcraft release` commands that'd be run instead of actually executing them.

I decided to push everything back to the beta channel rather than continuing to use candidate because it's what we did before, it is considered a less stable channel in the snap ecosystem, and we have way more beta users (~2.2k) than candidate users (2). By pushing our release to the beta channel, those users will essentially get the same experience as our stable users.

I already updated the release instructions and snapcraft credentials in Azure Pipelines.

- [ ] If this PR lands, I'll close the candidate channel which will cause those users to switch to stable until candidate is open again.

### CODE REVIEW

File: .azure-pipelines/templates/jobs/snap-deploy-job.yml [hunk 1]
```diff
@@ -30,7 +29,7 @@ jobs:
   # https://docs.microsoft.com/en-us/azure/devops/pipelines/library/secure-files?view=azure-devops#q-how-do-i-authorize-a-secure-file-for-use-in-a-specific-pipeline.
   #
   # This file has a maximum lifetime of one year and the current file will
-  # expire on 2023-09-06. The file will need to be updated before then to
+  # expire on 2024-02-10. The file will need to be updated before then to
```
Comment (wgreenberg) [inline .azure-pipelines/templates/jobs/snap-deploy-job.yml:32]: do we need to re-authorize this once merged?

**Review by wgreenberg — APPROVED** [review]
LGTM, just had the one question.

File: .azure-pipelines/templates/jobs/snap-deploy-job.yml (hunk 1 above)
Comment (bmw) [reply .azure-pipelines/templates/jobs/snap-deploy-job.yml:32]: Do you mean authorizing the new credentials for the "nightly" and "release" pipelines as described above?

If so, nope! I went ahead and updated the credentials last night. I had to reauthorize things as a part of that, but that's all been done.

**Comment by bmw** [comment]
Thanks for the quick review!

**Pull request merged**